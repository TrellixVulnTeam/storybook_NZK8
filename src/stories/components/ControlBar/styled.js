// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  background: ${colors.white};
`;

export const Section = styled.div`
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
  justify-content: left;
  align-items: center;
  padding: 0 30px;
  border-bottom: 1px solid ${colors.paleGrey};
`;

export const Tab = styled.div`
  font-size: 14px;
  color: ${p => {
    if (p.active) {
      if (p.theme === 'blue') {
        return colors.brightBlue;
      }
      return colors.darkGrey;
    }
    return colors.steel;
  }};
  font-weight: ${p => (p.active ? 600 : 'none')};
  transition: background-color 200ms ease;
  text-align: center;
  height: 48px;
  display: flex;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  border-bottom: ${p => {
    if (p.active) {
      if (p.theme === 'blue') {
        return `2px solid ${colors.brightBlue}`;
      }
      return `1px solid ${colors.strawberry}`;
    }
    return `1px solid ${colors.paleGrey}`;
  }};
  margin-bottom: -1px;

  &:hover {
    cursor: pointer;
    color: ${p => (!p.active ? 'null' : `${colors.brightBlue}`)};
    border-bottom: ${p => (p.active ? 'null' : `1px solid ${colors.slateGrey}`)};
  }

  @media (min-width: 1200px) {
  }
`;
