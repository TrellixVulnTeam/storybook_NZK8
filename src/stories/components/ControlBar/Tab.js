// @flow

import React from 'react';
import { Tab } from './styled';

import type { ControlBarItem } from './index';

type Props = {
  item: ControlBarItem,
  onSelectItem: ControlBarItem => void,
  theme?: string,
};

export default (props: Props) => {
  const active = Boolean(props.item.value);

  return (
    <Tab active={active} onClick={() => props.onSelectItem(props.item)} theme={props.theme}>
      {props.item.label}
    </Tab>
  );
};
