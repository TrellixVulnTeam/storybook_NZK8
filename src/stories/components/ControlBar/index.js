// @flow

import React, { PureComponent } from 'react';
import { Wrapper, Section } from './styled';

import Tab from './Tab';

type ItemType = 'tab' | 'dropdown-tab' | 'dropdown-nav' | 'checkbox';

type DropdownNavItem = {
  label: string,
  // eslint-disable-next-line no-use-before-define
  onSelectItem: ControlBarItem => void,
};

export type ControlBarItem = {
  type: ItemType,
  label: string,
  value: boolean | string,
  dropdownValues?: Array<string | DropdownNavItem>,
  data: any,
};

type Props = {
  onSelectItem: ControlBarItem => void,
  left?: ?Array<ControlBarItem>,
  center?: ?Array<ControlBarItem>,
  right?: ?Array<ControlBarItem>,
  className?: string,
  theme?: string,
};

class ControlBar extends PureComponent<Props> {
  renderDropdownToggle = () => null;

  renderDropdownNav = () => null;

  renderCheckbox = () => null;

  /**
   * Render list of control bar items
   * @param {Array<Object>} items
   * @return {React$Node}
   */
  renderItems = (items: Array<ControlBarItem>): Array<React$Node> =>
    items.map((item, i) => {
      if (item.type === 'tab') {
        // eslint-disable-next-line react/no-array-index-key
        return (
          <Tab
            // eslint-disable-next-line react/no-array-index-key
            key={`tab-${i}`}
            onSelectItem={this.props.onSelectItem}
            item={item}
            theme={this.props.theme}
          />
        );
      }

      if (item.type === 'dropdown-toggle') {
        return this.renderDropdownToggle();
      }

      if (item.type === 'dropdown-nav') {
        return this.renderDropdownNav();
      }

      if (item.type === 'checkbox') {
        return this.renderCheckbox();
      }

      return null;
    });

  /**
   * Renders a super-customizable control bar.
   * The control bar has three main sections:
   *   - left justified items
   *   - center items
   *   - right justified items
   * Each ControlBarItem can be a tab, a label with a dropdown value, and more
   * @param {Object} props
   * @return {React$Node}
   */
  render() {
    const { left, center, right, className } = this.props;

    return (
      <Wrapper className={className}>
        {left && left.length ? <Section>{this.renderItems(left)}</Section> : null}

        {center && center.length ? <Section>{this.renderItems(center)}</Section> : null}

        {right && right.length ? <Section>{this.renderItems(right)}</Section> : null}
      </Wrapper>
    );
  }
}

export default ControlBar;
