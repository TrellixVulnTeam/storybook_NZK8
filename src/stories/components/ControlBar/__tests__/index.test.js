import React from 'react';
import { mount } from 'enzyme';
import ControlBar from 'components/ControlBar/index';

describe('<ControlBar />', () => {
  it('should render correctly with full props', () => {
    const props = {
      onSelectItem: jest.fn(),
      left: [
        {
          type: 'tab',
        },
      ],
      right: [
        {
          type: 'dropdown-toggle',
        },
        {
          type: 'dropdown-nav',
        },
      ],
      center: [
        {
          type: 'checkbox',
        },
        {},
      ],
      className: 'className',
    };
    const sut = mount(<ControlBar {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with empty items', () => {
    const props = {
      onSelectItem: jest.fn(),
      left: [],
      right: [],
      center: [],
      className: 'className',
    };
    const sut = mount(<ControlBar {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
