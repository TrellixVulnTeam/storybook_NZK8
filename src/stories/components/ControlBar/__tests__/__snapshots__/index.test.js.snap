// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlBar /> should render correctly with empty items 1`] = `
<ControlBar
  center={Array []}
  className="className"
  left={Array []}
  onSelectItem={[MockFunction]}
  right={Array []}
>
  <styled__Wrapper
    className="className"
  >
    <div
      className="className styled__Wrapper-xs9hpj-0 fEvvEH"
    />
  </styled__Wrapper>
</ControlBar>
`;

exports[`<ControlBar /> should render correctly with full props 1`] = `
<ControlBar
  center={
    Array [
      Object {
        "type": "checkbox",
      },
      Object {},
    ]
  }
  className="className"
  left={
    Array [
      Object {
        "type": "tab",
      },
    ]
  }
  onSelectItem={[MockFunction]}
  right={
    Array [
      Object {
        "type": "dropdown-toggle",
      },
      Object {
        "type": "dropdown-nav",
      },
    ]
  }
>
  <styled__Wrapper
    className="className"
  >
    <div
      className="className styled__Wrapper-xs9hpj-0 fEvvEH"
    >
      <styled__Section>
        <div
          className="styled__Section-xs9hpj-1 lkkohE"
        >
          <_default
            item={
              Object {
                "type": "tab",
              }
            }
            key="tab-0"
            onSelectItem={[MockFunction]}
          >
            <styled__Tab
              active={false}
              onClick={[Function]}
            >
              <div
                className="styled__Tab-xs9hpj-2 aVytJ"
                onClick={[Function]}
              />
            </styled__Tab>
          </_default>
        </div>
      </styled__Section>
      <styled__Section>
        <div
          className="styled__Section-xs9hpj-1 lkkohE"
        />
      </styled__Section>
      <styled__Section>
        <div
          className="styled__Section-xs9hpj-1 lkkohE"
        />
      </styled__Section>
    </div>
  </styled__Wrapper>
</ControlBar>
`;
